extensions/resource-deployment/src/ui/toolsAndEulaSettingsPage.ts

```diff
@@ -234,9 +238,16 @@ export class ToolsAndEulaPage extends ResourceTypePage {
 					});
 				}
 
-				if (this._resourceType.agreements) {
-					this._agreementContainer.addItem(this.createAgreementCheckbox());
+				const agreementInfo = this._resourceType.getAgreementInfo(this.getSelectedOptions());
+				if (agreementInfo) {
+					this._agreementContainer.addItem(this.createAgreementCheckbox(agreementInfo));
```

Comment:
It looks like this was a separate bug fix where we'd add the checkbox even if something didn't have an agreement?

Reply:
The page breaks when one resource sub type has an agreement text but the other one doesn't. This was happening because we are doing null checks on agreement array instead of individual agreements.

Reply:
Next time please make sure to call out fixes like this - and more preferably separate them out into their own PR.